### PR TITLE
Fix torpedo store

### PR DESCRIPTION
**Pull request by radl97**
Fixed 3 bugs caught by SonarCloud.

Java Random class should not be instantiated only once.
Exception was forgotten to be thrown.
typo: =- was written instead of -=

### CODE REVIEW

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 1]
```diff
@@ -27,21 +27,20 @@ public TorpedoStore(int numberOfTorpedos){
...
     boolean success = false;
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
     double r = generator.nextDouble();
 
     if (r >= FAILURE_RATE) {
       // successful firing
-      this.torpedoCount =- numberOfTorpedos;
+      this.torpedoCount -= numberOfTorpedos;
```
Comment (annab97) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:45]: nice solution

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 2]
```diff
@@ -27,21 +27,20 @@ public TorpedoStore(int numberOfTorpedos){
       }
     }
   }
-
+  static Random generator = new Random();
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+      throw new IllegalArgumentException("numberOfTorpedos");
```
Comment (annab97) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:35]: clean code

**Review by annab97 — CHANGES_REQUESTED** [review]
Nice solution but there's still place for improvement!

**Review by radl97 — COMMENTED** [review]
I'm updating it now; please re-review it

**Review by annab97 — APPROVED** [review]
Nice work

**Pull request merged**